### PR TITLE
[kops] rbac support

### LINKED ISSUE #228: Enable Optional RBAC in Kops Manifest
## what
* Add env flag to enable/disable RBAC (default to off, for backwards compatibility)

## why
* RBAC is the new norm (for almost a year now)
* We need to start supporting it

### PR DESCRIPTION
**Pull request by alebabai**
## what
* beautify manifest template
* add rbac support (working example from here https://pracucci.com/kubernetes-rbac-with-kops.html)

## why
closes #228

### CODE REVIEW

**Comment by osterman** [comment]
@alebabai for reference, one our customers is using:
```
  kubeAPIServer: 
    authorizationMode: RBAC,AlwaysAllow 
    authorizationRbacSuperUser: cluster-admin 
```
Do we need this?

File: rootfs/templates/kops/default.yaml [hunk 1]
```diff
@@ -18,37 +18,45 @@ spec:
   api:
     loadBalancer:
       type: Public
+  {{- if bool (getenv "KOPS_AUTHORIZATION_RBAC_ENABLED") }}
+  kubeAPIServer:
+    authorizationRbacSuperUser: {{ getenv "KOPS_AUTHORIZATION_RBAC_SUPER_USER" | default "admin" }}
+  {{- end }}
   authorization:
+    {{- if bool (getenv "KOPS_AUTHORIZATION_RBAC_ENABLED") }}
+    rbac: {}
```
Comment (osterman) [inline rootfs/templates/kops/default.yaml]: See comments

**Comment by alebabai** [comment]
@osterman 
> Parameterize rbac section
what do you mean, what kind parameters?

**Comment by osterman** [comment]
What are the parameters to `rbac: {}`?

File: rootfs/templates/kops/default.yaml [hunk 2]
```diff
@@ -18,37 +18,45 @@ spec:
   api:
     loadBalancer:
       type: Public
+  {{- if bool (getenv "KOPS_AUTHORIZATION_RBAC_ENABLED") }}
+  kubeAPIServer:
```
Comment (osterman) [inline rootfs/templates/kops/default.yaml]: Do we need `authorizationMode`?

Comment (alebabai) [reply rootfs/templates/kops/default.yaml]: I don't think so, because it doesn't specified in cluster.spec

**Comment by alebabai** [comment]
> What are the parameters to rbac: {}?
it seems to me that there aren't any params here
https://github.com/kubernetes/kops/blob/master/docs/apireference/build/documents/_generated_authorizationspec_v1alpha2_kops_definition.md
https://github.com/kubernetes/kops/blob/master/docs/apireference/build/documents/_generated_rbacauthorizationspec_v1alpha2_kops_definition.md

File: rootfs/templates/kops/default.yaml (hunk 2 above)
Comment (osterman) [reply rootfs/templates/kops/default.yaml]: Bump

Comment (alebabai) [reply rootfs/templates/kops/default.yaml]: i read about `kubeAPIServer` props, so the only thing that could be usefull is `authorizationMode: RBAC,AlwaysAllow`. It explicitly declares authorization modes.

File: rootfs/templates/kops/default.yaml [hunk 3]
```diff
@@ -18,37 +18,46 @@ spec:
...
+    authorizationMode: RBAC,AlwaysAllow
```
Comment (osterman) [inline rootfs/templates/kops/default.yaml]: make this an env that defaults to `RBAC,AlwaysAllow`

Comment (alebabai) [reply rootfs/templates/kops/default.yaml]: done

**Pull request merged**